x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -264,10 +268,21 @@ export class AlertingPlugin {
     savedObjects: SavedObjectsServiceStart,
     elasticsearch: ElasticsearchServiceStart
   ): (request: KibanaRequest) => Services {
-    return request => ({
-      callCluster: elasticsearch.legacy.client.asScoped(request).callAsCurrentUser,
-      savedObjectsClient: savedObjects.getScopedClient(request),
-    });
+    return request => {
+      const caller = elasticsearch.legacy.client.asScoped(request).callAsCurrentUser;
+      return {
+        callCluster: caller,
+        savedObjectsClient: savedObjects.getScopedClient(request),
+        search: this.data!.search.createScopedSearchApi(caller).search,
+        indexPattern: {
+          async getById(id: string) {
+            // not implemented, we need changes in the data plugin
+            // this is just to express what we'd hbroadly need access to
+            return undefined;
+          },
```

Comment:
We'll need App-Arch's help here as we need access to the IndexPattern (it's needed to convert a SavedQuery to an ES DSL query) and that is currently encapsulated in `data` and is only available on the Front End.